src/main/java/org/optaplanner/springboottaskassigning/domain/TaskAssigningSolution.java

```diff
@@ -18,33 +18,76 @@
 
 import java.util.List;
 
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToMany;
+
 import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import org.hibernate.annotations.Columns;
+import org.hibernate.annotations.LazyCollection;
+import org.hibernate.annotations.LazyCollectionOption;
+import org.hibernate.annotations.Parameter;
+import org.hibernate.annotations.TypeDef;
 import org.optaplanner.core.api.domain.solution.PlanningEntityCollectionProperty;
 import org.optaplanner.core.api.domain.solution.PlanningScore;
 import org.optaplanner.core.api.domain.solution.PlanningSolution;
 import org.optaplanner.core.api.domain.solution.drools.ProblemFactCollectionProperty;
 import org.optaplanner.core.api.domain.valuerange.ValueRangeProvider;
 import org.optaplanner.core.api.score.buildin.bendable.BendableScore;
+import org.optaplanner.persistence.jpa.impl.score.buildin.bendable.BendableScoreHibernateType;
 
 @PlanningSolution
 @JsonPropertyOrder({"skillList", "taskTypeList", "customerList", "employeeList", "taskList", "score", "frozenCutoff"})
+@Entity
```

Comment:
Using the class as:
- Planning solution
- JSON serialization entity
- JPA entity
are actually three different responsibilities.

Reply:
Yes, but we want to allow users to combine those into just one class.

That being said, there will be plenty of users that architect some of these responsibility in different classes. For example, those that follow the Data Transfer Object pattern.

Reply:
Given it's an example, I am fine with that.